docs/user/rstudio/ide/guide/tools/copilot.qmd

```diff
@@ -0,0 +1,237 @@
+
+## Setup
+
+To enable GitHub Copilot in RStudio:
+
+1.  Navigate to Tools \> Global Options \> Copilot.  \
+2.  Check the box to "Enable GitHub Copilot".  \
+3.  Download and install the Copilot Agent components.  \
+4.  Click the "Sign In" button.  \
+5.  In the "GitHub Copilot: Sign in" dialog, copy the Verification Code.  \
+
+![](images/copilot_sign-in.png){fig-alt="GitHub Copilot: Sign in"}
+
+6.  Navigate to or click on the link to <https://github.com/login/device>, paste the Verification Code and click "Continue".  \
+7.  GitHub will request the necessary permissions for GitHub Copilot. To approve these permissions, click "Authorize GitHub Copilot Plugin".  
+8. After the permissions have been approved, your RStudio IDE will indicate the currently signed in user.  
+5.  Close the Global Options dialogue, open a source file (`.R`, `.py`, `.qmd`, etc) and begin coding with Copilot  
+
+To disable GitHub Copilot either:
+
+-   Navigate to Tools \> Global Options \> Copilot and uncheck "Enable GitHub Copilot"\
+-   Use the Command Palette <kbd>Cmd+Shift+P</kbd> on Mac or <kbd>Ctrl+Shift+P</kbd> on Windows/Linux, search for "Enable GitHub Copilot" and then uncheck "Enable GitHub Copilot"
+
+## Using Copilot
+
+Copilot offers autocomplete-style suggestions as you code as "ghost text". This ghost-text is similar to the existing autocomplete available in RStudio but importantly is a generated suggestion rather than an exact auto-completion.
+
+![](images/copilot-rstudio.png){fig-alt="Copilot in RStudio"}
+
+1.  A simple but specific comment providing additional context to Copilot
+2.  Copilot suggestion, shown in light grey 'ghost text'
+3.  The Copilot status bar, which indicates whether RStudio is waiting on a response to be generated, a completion response has been received, or no completions are available.
+
+GitHub Copilot primarily relies on the context in the file you are actively editing. Any comments, code, or other context provided within the active document will be used as a "prompt" that Copilot will then use to provide a suggested completion. To expand the scope of the context used by Copilot beyond just the active document, there is a setting to also index and read from other R, Python, or SQL files in the current project. This setting can be toggled on or off in the Tools \> Global Options \> Copilot \> "Index project files with GitHub Copilot" setting.
+
+For general advice on how to use Copilot, please see:\
+- [How to use GitHub Copilot: Prompts, tips, and use cases](https://github.blog/2023-06-20-how-to-write-better-prompts-for-github-copilot/)\
+- [How to use GitHub Copilot](https://docs.github.com/en/copilot/getting-started-with-github-copilot/how-to-use-github-copilot)\
```

Comment:
The URL doesn't work. Is it supposed to be https://docs.github.com/en/copilot/getting-started-with-github-copilot ?

Reply:
Hmm - I don't think that's right. I may exclude that second URL altogether.

Reply:
Removed second url